druid/examples/calc.rs

```diff
@@ -165,44 +178,79 @@ fn digit_button(digit: u8) -> impl Widget<CalcState> {
         .on_click(move |_ctx, data: &mut CalcState, _env| data.digit(digit))
 }
 
-fn flex_row<T: Data>(
+fn flex_row3<T: Data>(
+    w1: impl Widget<T> + 'static,
+    w2: impl Widget<T> + 'static,
+    w3: impl Widget<T> + 'static,
+) -> impl Widget<T> {
+    Flex::row()
+        .with_flex_child(Padding::new(1., w1), 2.0)
+        .with_flex_child(Padding::new(1., w2), 1.0)
+        .with_flex_child(Padding::new(1., w3), 1.0)
+}
+
+fn flex_row4<T: Data>(
     w1: impl Widget<T> + 'static,
     w2: impl Widget<T> + 'static,
     w3: impl Widget<T> + 'static,
     w4: impl Widget<T> + 'static,
 ) -> impl Widget<T> {
     Flex::row()
-        .with_flex_child(w1, 1.0)
-        .with_spacer(1.0)
-        .with_flex_child(w2, 1.0)
-        .with_spacer(1.0)
-        .with_flex_child(w3, 1.0)
-        .with_spacer(1.0)
-        .with_flex_child(w4, 1.0)
+        .with_flex_child(Padding::new(1., w1), 1.0)
+        .with_flex_child(Padding::new(1., w2), 1.0)
+        .with_flex_child(Padding::new(1., w3), 1.0)
+        .with_flex_child(Padding::new(1., w4), 1.0)
 }
 
 fn build_calc() -> impl Widget<CalcState> {
+    let display_painter = Painter::new(|ctx, _, env| {
+        let bounds = ctx.size().to_rect();
+        if let Ok(brush) = ctx.gradient(FixedLinearGradient {
```

Comment:
Is there a reason this gradient can't be made with `LinearGradient::new`?

Reply:
```
204 |         if let Ok(brush) = ctx.gradient(LinearGradient::new(
    |                                ^^^^^^^^ the trait `std::convert::From<piet::gradient::LinearGradient>` is not implemented for `piet::gradient::FixedGradient`
```

Maybe i missed something, but it didn't work.